builtin/providers/librato/resource_librato_metric_test.go

```diff
@@ -0,0 +1,230 @@
+					resource.TestCheckResourceAttr(
+						"librato_metric.foobar", "name", name),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckLibratoMetricDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*librato.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "librato_metric" {
+			continue
+		}
+
+		fmt.Printf("*** testAccCheckLibratoMetricDestroy ID: %s\n", rs.Primary.ID)
+		_, _, err := client.Metrics.Get(rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Metric still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckLibratoMetricName(metric *librato.Metric, name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if metric.Name == nil || *metric.Name != name {
+			return fmt.Errorf("Bad name: %s", *metric.Name)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckLibratoMetricType(metric *librato.Metric, validTypes []string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		m := make(map[string]bool)
+		for _, v := range validTypes {
+			m[v] = true
+		}
+
+		if !m[*metric.Type] {
+			return fmt.Errorf("Bad metric type: %s", *metric.Type)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckLibratoMetricExists(n string, metric *librato.Metric) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Metric ID is set")
+		}
+
+		client := testAccProvider.Meta().(*librato.Client)
+
+		foundMetric, _, err := client.Metrics.Get(rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		if foundMetric.Name == nil || *foundMetric.Name != rs.Primary.ID {
+			return fmt.Errorf("Metric not found")
+		}
+
+		*metric = *foundMetric
+
+		return nil
+	}
+}
+
+func counterMetricConfig(name, typ, desc string) string {
+	return strings.TrimSpace(fmt.Sprintf(`
+    resource "librato_metric" "foobar" {
+        name = "%s"
+        type = "%s"
+        description = "%s"
+        attributes {
+          display_stacked = true,
+          created_by_ua = "go-librato/0.1"
+        }
+    }`, name, typ, desc))
+}
+
+func gaugeMetricConfig(name, typ, desc string) string {
+	return strings.TrimSpace(fmt.Sprintf(`
+    resource "librato_metric" "foobar" {
+        name = "%s"
+        type = "%s"
+        description = "%s"
+        attributes {
+          display_stacked = true,
+          created_by_ua = "go-librato/0.1"
+        }
+    }`, name, typ, desc))
+}
+
+func compositeMetricConfig(name, typ, desc string) string {
+	return strings.TrimSpace(fmt.Sprintf(`
+    resource "librato_metric" "foobar" {
+        name = "%s"
+        type = "%s"
+        description = "%s"
+        composite = "s(\"librato.cpu.percent.user\", {\"environment\" : \"prod\", \"service\": \"api\"})"
+        attributes {
+          display_stacked = true,
+          created_by_ua = "go-librato/0.1"
```

Comment:
I don't think anyone would ever write this attribute in a config; presumably it's set by go-librato when the metric is created. Does the resource behave as expected without this in the config?